### ISSUE TITLE
WebhooksManager#current_webhooks raises NoMethodError: undefined method `index_by' for nil:NilClass

### ISSUE THREAD

**Issue by rochlefebvre** [Shopify/shopify_app#700]
Encountered with shopify_app-8.3.2 in a rails 5.2.2/ruby 2.6.0 application.
```
{
  "class": "ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper",
  "wrapped": "ShopifyApp::WebhooksManagerJob",
  # ...
  "error_message": "undefined method `index_by' for nil:NilClass",
  "error_class": "NoMethodError",
  "failed_at": 1551291482.9294562,
  "retry_count": 3,
  "error_backtrace": [
    "/app/vendor/bundle/ruby/2.6.0/gems/shopify_app-8.3.2/lib/shopify_app/managers/webhooks_manager.rb:58:in `current_webhooks'",
    "/app/vendor/bundle/ruby/2.6.0/gems/shopify_app-8.3.2/lib/shopify_app/managers/webhooks_manager.rb:54:in `webhook_exists?'",
    "/app/vendor/bundle/ruby/2.6.0/gems/shopify_app-8.3.2/lib/shopify_app/managers/webhooks_manager.rb:28:in `block in create_webhooks'",
    "/app/vendor/bundle/ruby/2.6.0/gems/shopify_app-8.3.2/lib/shopify_app/managers/webhooks_manager.rb:27:in `each'",
    "/app/vendor/bundle/ruby/2.6.0/gems/shopify_app-8.3.2/lib/shopify_app/managers/webhooks_manager.rb:27:in `create_webhooks'",
    "/app/vendor/bundle/ruby/2.6.0/gems/shopify_app-8.3.2/lib/shopify_app/jobs/webhooks_manager_job.rb:11:in `block in perform'",
    "/app/vendor/bundle/ruby/2.6.0/gems/shopify_api-5.2.0/lib/shopify_api/session.rb:30:in `temp'",
    "/app/vendor/bundle/ruby/2.6.0/gems/shopify_app-8.3.2/lib/shopify_app/jobs/webhooks_manager_job.rb:9:in `perform'",
    "/app/vendor/bundle/ruby/2.6.0/gems/activejob-5.2.2/lib/active_job/execution.rb:39:in `block in perform_now'",
```
The error is raised from this statement:
https://github.com/Shopify/shopify_app/blob/363a5e3fe30377257e9907531ca44d66c52f6080/lib/shopify_app/managers/webhooks_manager.rb#L57-L59

The inheritance chain for Webhook is ShopifyAPI::Webhook < ShopifyAPI::Base < ActiveResource::Base

The [`all`](https://github.com/rails/activeresource/blob/master/lib/active_resource/base.rb#L1004) method on ActiveResource::Base:
```ruby
# This is an alias for find(:all). You can pass in all the same
# arguments to this method as you can to <tt>find(:all)</tt>
def all(*args)
  find(:all, *args)
end
```
```ruby
# https://github.com/rails/activeresource/blob/master/lib/active_resource/base.rb#L969
def find(*arguments)
  scope   = arguments.slice!(0)
  options = arguments.slice!(0) || {}

  case scope
  when :all
    find_every(options)
```
```ruby
# https://github.com/rails/activeresource/blob/master/lib/active_resource/base.rb#L1062
def find_every(options)
  begin
   # ...
  rescue ActiveResource::ResourceNotFound
    # Swallowing ResourceNotFound exceptions and return nil - as per
    # ActiveRecord.
    nil
  end
end
```
So it should be expected that `Webhook.all` _can_ return `nil`.

**Comment by Alex1sz** [comment]
+ 1 maybe update the Webhook documentation to indicate that the webhook functionality is broken and buggy so people know not to use it

**Comment by mkevinosullivan** [comment]
Fixed in Shopify/shopify_app#720
